openapi3gen/openapi3gen_test.go

```diff
@@ -590,3 +590,32 @@ func ExampleNewSchemaRefForValue_recursive() {
 	//   "type": "object"
 	// }
 }
+
+type ID [16]byte
+
+// T implements SetSchemar, allowing it to set an OpenAPI schema.
+type T struct {
+	ID ID `json:"id"`
+}
+
+func (_ *ID) SetSchema(schema *openapi3.Schema) {
+	schema.Type = &openapi3.Types{"string"} // Assuming this matches your custom implementation
+	schema.Format = "uuid"
+}
+func TestNewSchemaRefForValueWithSetSchemar(t *testing.T) {
```

Comment:
Please turn this into an example so usage of `SetSchema` gets examples in the docs

This name example's name should include the name of the interface you introduce.

https://go.dev/blog/examples

Reply:
hi @fenollp thank you for the review! i just change my `Test` to use `Example` and already put the output also. kindly re-review it again. thank you